[WIP] [DOC-10641] Add reference for encode-uri (#18993)
cockroachdb/docs: src/current/v24.3/cockroach-encode-uri.md

---
title: cockroach encode-uri
summary: Get the connection string for a cluster from its URI.
toc: true
docs_area: reference.cli
---

> The `cockroach encode-uri` command returns the `postgres://` connection string for a cluster's `defaultdb` database from the URI of a node.

Author: ```suggestion
The `cockroach encode-uri` command returns the `postgresql:/` connection string for a cluster's `defaultdb` database from the URI of a node.
```

> The `cockroach encode-uri` command returns the `postgresql:/` connection string for a cluster's `defaultdb` database from the URI of a node.

Author: ```suggestion
The `cockroach encode-uri` command returns the `postgresql:/` connection string for a cluster and database from the URI of a cluster node.
```

## Synopsis

To get the connection string for a cluster's `defaultdb` database, replace `{URI}` with the URI of a cluster node.

~~~ shell

> cockroach encode-uri {URI}

Author: Can you verify that the protocol here expects two slashes? It looks like the returned URI contains only a single slash.

~~~

To specify the username to use in the connection string, replace `{USERNAME}` with a SQL username.

{% include_cached copy-clipboard.html %}
~~~ shell
cockroach encode-uri --user {USERNAME} {URI}
~~~

To specify the database to use in the connection string, replace `{DATABASE}` with the name of a database.

{% include_cached copy-clipboard.html %}
~~~ shell
cockroach encode-uri --database {DATABASE} {URI}
~~~

## Flags

The `encode-uri` command supports the following flags. All flags are optional.

Flag          | Description
--------------|-----------

> `--inline`    | Whether to inline certificates for [Physical Cluster Replication]({% link {{ page.version.version }}/physical-cluster-replication.md %}). Defaults to `false`.

Author: ```suggestion
`--inline`    | Whether to inline certificates for [Physical Cluster Replication]({% link {{ page.version.version }}/physical-cluster-replication-overview.md %}). Defaults to `false`.
```

---
title: cockroach encode-uri
summary: Construct a connection URI that a client or application can use to connect to a cluster, based on the command-line arguments you provide.
toc: true
docs_area: reference.cli
---

> The `cockroach encode-uri` command constructs a connection URI that a client or application can use to connect to a cluster, based on the command-line arguments you provide. The returned connection URI is in `postgresql:/` format.

Reviewer: ```suggestion
The `cockroach encode-uri` command constructs a connection URI that a client or application can use to connect to a cluster, based on the command-line arguments you provide. The returned connection URI is in `postgresql://` format.
```

We only ever want to refer to the scheme as either `postgresql` or `postgresql://`. `postgres:/` will not do what the user thinks.

## Synopsis

{% include_cached copy-clipboard.html %}
~~~ shell
cockroach encode-uri [postgres:/][{USERNAME}[:{PASSWORD}]@]{HOST[:{PORT}]} [{FLAGS}]
~~~

Only `{HOST}` is required; all other options and [flags](#flags) are optional, as indicated by square brackets.

## Options

> - `postgres:/`: Optional protocol specifier if providing a connection URI as input, rather than a `HOST:PORT` combination.

Reviewer: ```suggestion
- `postgres://`: Optional postgresql URI scheme.
```
